L50-cpp-sts-creds-api.md

```diff
@@ -0,0 +1,89 @@
+Title
+----
+* Author(s): Julien Boeuf
+* Approver: a11r
+* Status: Draft
+* Implemented in: core, cpp
+* Last updated: May 20, 2019
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+This is a proposal to add support for a new type of credentials in the cpp stack implementing the [OAuth 2.0 Token Exchange](https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-16) which documents a protocol for an HTTP- and JSON- based Security Token Service (STS) by defining how to request and obtain security tokens from OAuth 2.0 authorization servers, including security tokens employing impersonation and delegation.
+
+## Background
+
+[OAuth 2.0 Token Exchange](https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-16) is the proposed standard for exchanging tokens, for scenarios where a client needs to access a resource that does not natively accept the credentials created by the authentication system used by the client. In these settings, enabling credential exchange can provide better security: Otherwise, the client needs to acquire a different set of credentials to access the resource, and the new credentials must be managed securely.
+
+For instance, Kubernetes workloads can securely obtain credentials from their platform, and they can use these to authenticate to other entities in the cluster, or to the Kubernetes API itself. However, if a Kubernetes workload requires access to a cloud provider's API, it needs a separate set of secrets, and a way to deploy these secrets securely. With credential exchange this is no longer necessary: The workload credential can be directly exchanged for a short-lived token for the cloud API. We expect this type of interaction to become a best practice, as it eliminates a number of problematic patterns, such as secrets checked into code repos, or stored in improperly ACL'ed files.
```

Comment:
Minor: Remove the comma in "code repos, or stored"?

Reply:
done.